enterprise/backend/src/metabase_enterprise/audit_db.clj

```diff
@@ -83,25 +85,30 @@
   "Default Dashboard Overview (this is a dashboard) entity id."
   "bJEYb0o5CXlfWFcIztDwJ")
 
-(defn entity-id->object
+(def ^{:arglists '([audit-installed? model entity-id])
+       :private  true} memoized-select-audit-entity*
+  (mdb/memoize-for-application-db
+   (fn [audit-installed? model entity-id]
+     (when audit-installed?
+       (t2/select-one model :entity_id entity-id)))))
+
+(defn memoized-select-audit-entity
```

Comment:
1. this was not memorized at all before
2. and since it's now memorized, I think it's nice to make sure it takes the state of `audit-installed` into account to prevent cases when we accidentally call these selectors before audit is installed. I noticed we use this pattern in test [here](https://github.com/metabase/metabase/blob/e6e3f85aeb0d84b06b6963543624d3aa6958ce1a/enterprise/backend/test/metabase_enterprise/audit_app/permissions_test.clj#L124)

Reply:
Does the name need to change though? 😁

Reply:
the old name is quite generic IMO, and since we added `audit-installed` as a param so I think we need to make it more specific